api/api_public_blockchain.go

```diff
@@ -295,6 +295,8 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	// this makes sure resources are cleaned up.
 	defer cancel()
 
+	// Add gas fee to sender for estimating gasLimit/computing cost or calling a function by insufficient balance sender.
+	state.AddBalance(msg.ValidatedSender(), new(big.Int).Mul(new(big.Int).SetUint64(msg.Gas()), msg.GasPrice()))
```

Comment:
Before adding this, what happen when sender does not have enough balance?

Reply:
@jimni1222 
I moved this line from [here](https://github.com/klaytn/klaytn/pull/1134/files#diff-c7deb6fb16afa13f94bea51c3e7d8d9da0a069beb20036f94a6bae90442a499dL209-L210).

It because now EthDoCall (used by eth_call) and DoCall (used by klay_call) both using GetEVM and works differently.
* EthDoCall requires caller have enough gas to call.
* DoCall requires no gas to call.

If we keep that line in GetEVM, it is hard to support that differences.